src/bsd.rs

```diff
@@ -0,0 +1,42 @@
+use std::fs::read_link;
+use std::path::PathBuf;
+use libc::*;
+use std::ptr;
+
+// http://stackoverflow.com/q/799679
+// http://stackoverflow.com/a/1024937
+pub fn get_executable_path() -> Option<PathBuf> {
+    // FreeBSD
+    let mib = [CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1];
+    let mut buf: Vec<u8> = Vec::with_capacity(1024);
+    let mut cb = 1024 as size_t;
+    let result = unsafe {
+        sysctl(mib.as_ptr(),
```

Comment:
OK, I'll do that once I have my different BSD environments running. FreeBSD will probably need an entirely different function since it uses either sysctl or a /proc file. Either way, I'll update the PR once I've pushed these modifications. Thanks for your feedback !